packages/hierarchies/src/hierarchies/imodel/FilteringHierarchyDefinition.ts

```diff
@@ -219,14 +307,18 @@ async function matchFilters<
           // ideally, `predicate` would act as a type guard to guarantee that `id` is `TIdentifier`, but at the moment
           // async type guards aren't supported
           id: nodeId as TIdentifier,
-          childrenIdentifierPaths: [],
           isFilterTarget: false,
         };
         matchingFilters.push(entry);
       }
-      const remainingPath = path.slice(1);
-      if (remainingPath.length > 0) {
-        entry.childrenIdentifierPaths.push(options ? { path: remainingPath, options } : remainingPath);
+
+      if (path.length > 1) {
+        if (HierarchyNodeIdentifier.isGenericNodeIdentifier(nodeId)) {
```

Comment:
why do we only need to do this for generic nodes from now on?

Reply:
Because we are not using `childrenIdentifierPaths` for `InstanceNodeIdentifiers`. Changed the implementation, now `matchFilters` function takes in another argument: `extractChildrenIdentifierPaths` which determines whether or not to retrieve `childrenIdentifierPaths`